flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/DefaultExecutionGraph.java

```diff
@@ -1279,10 +1279,14 @@ private void releasePartitions(final List<IntermediateResultPartitionID> releasa
 
             // Remove cached ShuffleDescriptor when partition is released
             releasablePartitions.stream()
-                    .map(IntermediateResultPartitionID::getIntermediateDataSetID)
+                    .map(edgeManager::getConsumedPartitionGroupsById)
```

Comment:
Could you help me understand why it is possible that an `IntermediateResultPartitionID` belongs to multiple `ConsumerPartitionGroups`?

Reply:
Thank you for reviewing this pull request, @tillrohrmann! Since a IntermediateDataSet can be consumed by multiple consumer JobVertices, the IntermediateResultPartition corresponding to this data set can be consumed by multiple consumer ExecutionVertices. These ExecutionVertices belong to different ConsumerVertexGroups. A ConsumerVertexGroup is corresponding to a ConsumedPartitionGroup. Therefore we can say an IntermediateResultPartition can belong to multiple ConsumedPartitionGroups. An example is illustrated in the figure below:
![2](https://user-images.githubusercontent.com/6576831/130045874-0a9e9432-3db2-4bdb-9e5a-eaafd7e9f304.jpg)
Although, currently there has to be only one consumer for each IntermediateDataSet by design.

Reply:
Hmm, why do support something that is currently not possible and might never be?

Reply:
I think you are right! @tillrohrmann 
We can simplify `getConsumedPartitionGroupsById` to  `getConsumedPartitionGroupById` and guard it with check that a partition must not be assigned to multiple `ConsumedPartitionGroup`s. I would not require much effort I believe.
Later we can do the refactor work to get rid of the assumption that one `IntermediateDataSet` can have multiple consumer `JobVertex` to further simplify things (FLINK-18295).